### ISSUE TITLE
mode.addVar(self, name='', vtype='C', lb=0.0, ub=None, obj=0.0, pricedVar = False)

### ISSUE THREAD

**Issue by fengshaoA** [scipopt/PySCIPOpt#621]
Let's say x = mode.addVar(self, name='', vtype='C', lb=0.0, ub=None, obj=5.0, pricedVar = False),then I can say that the initial value of x is 5.0?

**Comment by Joao-Dionisio** [comment]
No, you cannot say that. The obj parameter is the coefficient of the variable in the objective function. Suppose you had the following:

```
y = model.addVar(name="y", vtype="C", lb=0.0, pricedVar = False)
model.setObjective(3*y)
```
Here your objective function is min. 3*y. If you then do

```
x = model.addVar(name="x", vtype="C", lb=0.0, obj=5.0, pricedVar=False)
```
You do two things. First, you create and add a new variable x, and then add it to the objective function. So now your new objective function is: min 3*y + 5*x.

Please let me known if something is not clear.

**Comment by fengshaoA** [comment]
> 
Ok, I see, thank you very much, but what if I want to use the solution of the heuristic algorithm as the initial value in the scip solver? i know that in gurobi you can do this: for example, if the heuristic algorithm x[i] has a feasible solution of 5, then in gurobi you can set the initial value of x[i] like this: x[i].Start = 5;

**Comment by Joao-Dionisio** [comment]
If I understood correctly you want to initialize a partial solution, right? I have never done this, but I'm fairly certain that I got it. You first need to create a partial solution with [model.createPartialSol()](https://scipopt.github.io/PySCIPOpt/docs/html/classpyscipopt_1_1scip_1_1Model.html#a3915fe9a9e33f6edfeb6549a124b9b41), and then add values to the variables you want. This method (createPartialSol) also has an optional parameter for the heuristic you used to find the values. 

```
model = Model()
y = model.addVar(name="y",lb=3)
x = model.addVar(name="x",lb=7)

s = model.createPartialSol()
model.setSolVal(s, x, 50)

model.addCons(x >= y**2)

model.setObjective(x+y)
model.addSol(s)
model.optimize()
```

I made a non-linear model so that the trivial heuristic couldn't immediately find the solution. If you run this script, you will see that the first line is

> feasible solution found by completesol heuristic after 0.0 seconds, objective value 4.870000e+01

And the optimal value is 1.2e+01, so I'm fairly certain that this is the way to do what you want.

**Comment by fengshaoA** [comment]
Ok, thank you for your advice, I have revised the title, I will try your way, thank you again!

**Comment by fserra** [comment]
you mean that the version of SCIP that you are using is 6.0.2? You should update to the latest SCIP and the latest version of pyscipopt and try again

**Comment by fengshaoA** [comment]
> you mean that the version of SCIP that you are using is 6.0.2? You should update to the latest SCIP and the latest version of pyscipopt and try again

Ok, thanks, I've thought of that, and now I've updated to version 7.0.2, which includes createPartialSol,

**Comment by fengshaoA** [comment]
When I provide initial values for the model during the solution process, the solver log prompts me with this issue：
WARNING: ignore partial solution (0) because unknown rate is too large (0.996856 > 0.85)
After the prompt for the issue, the initial values I provided did not take effect. Could you please explain what is going on here?
![image](https://github.com/scipopt/PySCIPOpt/assets/24955395/98b9b4f4-f2f7-4433-80e9-d720a37b9aaa)
![image](https://github.com/scipopt/PySCIPOpt/assets/24955395/839c95f5-b4be-4e5b-8e15-1fa4e8776bd9)

**Comment by sbolusani** [comment]
The completesol heuristic of SCIP, which takes a partial solution as input and completes it, has a maximum threshold of 0.85 (85%) for the number of unknown variable values. Since the provided partial solution seems to have 0.996856 (more than 99.6%) of the unknown variable values, the heuristic printed the above warning message and exited without trying to create a complete solution.

You may either provide a partial solution with more known variable values (recommended option) or change the threshold value of the completesol heuristic by setting the following SCIP parameter. This parameter takes values in the range [0, 1] with a default value of 0.85 (hence, the above warning message with 0.85 in it).

`heuristics/completesol/maxunknownrate`
